### PR TITLE
Improve component support documentation

### LINKED ISSUE #316: Update the documentation for v1.4.0
**Feature Category**

- [ ] Correctness
- [x] User Interface / User Experience
- [ ] Performance
- [x] Other (please explain)

**Describe the problem**

Update the documentation. Especially, but not limited to:

- [ ] components of mquery, and how they interact in multi-server setting.
- [ ] how to write good yara rules (and link that from error in #315).

### PR DESCRIPTION
**Pull request by msm-code**
**Your checklist for this pull request**
- [ ] I've read the [contributing guideline](https://github.com/CERT-Polska/mquery/blob/master/CONTRIBUTING.md).
- [ ] I've tested my changes by building and running mquery, and testing changed functionality (if applicable)
- [ ] I've added automated tests for my change (if applicable, optional)
- [x] I've updated documentation to reflect my change (if applicable)

**What is the current behaviour?**
There is no explicit documentation of mquery components

**What is the new behaviour?**
There is a basic documentation of mquery components and what they are responsible for.

**Test plan**
n/a

**Closing issues**

fixes #316

### CODE REVIEW

File: docs/components.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+The workhorse of the entire setup. There must be at least one daemon for
+every ursadb instance. Daemon's tasks include querying the assigned ursadb
+instance for samples and running YARA rules on candidate samples.
+
+### Redis
+
+Is just a shared database used for communication between daemon and mquery.
+It is also used as a task queue for jobs scheduled for agents. And it's also
+used for persistent storage of job results. Finally, it's used to store
```
Comment (nazywam) [inline docs/components.md:44]:
```suggestion
used for persistent storage of job results. Finally, it's utilized to store
```

File: docs/components.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+plugin configuration and job cache for agents. In short, it's pretty overloaded
```
Comment (nazywam) [inline docs/components.md:45]:
```suggestion
plugin configuration and job cache for agents. To sum up, it's pretty overloaded
```

File: docs/distributed.md [hunk 3]
```diff
@@ -16,5 +16,6 @@ be connected to multiple mquery servers:
 ![](./distribured-rev.png)
 
 All of mquery's core functionality works in this setup, but there are some
-rough edges. Most notably, HTTP file download is hard to get working
-correctly (but hashes are computed and shown correctly).
+deployment problems. For example, web interface assumes that the samples are stored
+(or monuted) at the same location as in the workers. If that's not the case,
```
Comment (nazywam) [inline docs/distributed.md:20]:
```suggestion
(or mounted) at the same location as in the workers. If that's not the case,
```

File: docs/security.md [hunk 4]
```diff
@@ -14,23 +14,25 @@ and disallow anonymous users or at least don't give them admin rights.
 
 ### Mquery daemon (agent)
 
-No special considerations.
+No special considerations. Every daemon process must have network
+access to Redis and Ursadb.
```
Comment (nazywam) [inline docs/security.md:18]:
```suggestion
access to Redis and UrsaDB.
```

File: docs/security.md [hunk 5]
```diff
@@ -14,23 +14,25 @@ and disallow anonymous users or at least don't give them admin rights.
...
+the Redis database is necessary. There is no support for securing Redis
+with a password in the current version, so network isolation is
+the only way to prevent attacks. Most importantly, Redis must not
+be available from the public network.
 
 ### Ursadb
 
 Mquery daemons must have network access to their respective ursadb instances.
-Similarly to Redis, it's best to restrict network access to the ursadb instance. Ursadb protocol does not take malicious actors into account, and
+Similarly to Redis, it's best to restrict network access to the Ursadb instance. Ursadb protocol does not take malicious actors into account, and
```
Comment (nazywam) [inline docs/security.md:31]:
```suggestion
Similarly to Redis, it's best to restrict network access to the UrsaDB instance. Ursadb protocol does not take malicious actors into account, and
```

:see_no_evil:

File: docs/security.md [hunk 6]
```diff
@@ -14,23 +14,25 @@ and disallow anonymous users or at least don't give them admin rights.
...
 unauthenticated users can, for example, remove indexed data from the database,
 or cause a denial of service.
 
-In the provided docker-compose files, the ursadb user is overridden to root by
+In the provided docker-compose files, the Ursadb user is overridden to root by
```
Comment (nazywam) [inline docs/security.md:35]:
```suggestion
In the provided docker-compose files, the UrsaDB user is overridden to root by
```

**Review by nazywam — CHANGES_REQUESTED** [review]

File: docs/distributed.md (hunk 3 above)
Comment (msm-code) [reply docs/distributed.md:20]: huh why did my boy grammarly didn't catch this

**Comment by msm-code** [comment]
Someday I should sit down and fix all the naming inconsistencies :thinking:.

This also includes things like:
 - `Mquery` vs `mquery` at the start of sentence (I really dislike starting sentences from a non-capital letter though. Looks wrong to me)
 - `agent` vs `daemon` vs `worker` (this includes changes in the source code)
 - `taint` vs `tag` (this includes changes in ursadb)
 - probably others?

**Review by nazywam — APPROVED** [review]

**Pull request merged**